projects/kafka-cron/producer/internal/scheduler/scheduler.go

```diff
@@ -0,0 +1,35 @@
+package scheduler
+
+import (
+	"fmt"
+	"kafka-cron/pkg/models"
+	"kafka-cron/producer/internal/producer"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/robfig/cron/v3"
+)
+
+func Scheduler(p *kafka.Producer, jobs []models.CronJob, topic string) error {
+	c := cron.New(cron.WithSeconds())
+
+	// Schedule the jobs
+	for _, job := range jobs {
+		_, err := c.AddFunc(job.Schedule, func() {
+			err := producer.ProduceMessage(p, topic, job)
+			if err != nil {
+				fmt.Printf("error producing message: %v", err)
+			}
+		})
+		if err != nil {
+			return fmt.Errorf("error scheduling job: %v", err)
+		}
+	}
+
+	c.Start()
+
+	time.Sleep(5 * time.Minute)
+
+	c.Stop()
+	return nil
```

Comment:
I'm assuming this is just demo code while you're getting something working, but this kind of "How long should I keep going" code tends to be something we'd either put in `main` rather than in this function, or we'd pass in a function parameter for how long to keep going for. Having a function which blocks while it does something in the background for a period of time and then stops is a little weird :)